File: duplicate-assert/CSharpXUnit/CSharpXUnit.Tests/Tests/PostFilteringTargetWrapperOnlyDefaultFilter.cs
using Xunit;

namespace DefaultNamespace
{
    public class PostFilteringTargetWrapperTests
    {
        [Fact]
        public void PostFilteringTargetWrapperOnlyDefaultFilter()
        {
            {
                var target = new MyTarget();
                var wrapper = new PostFilteringTargetWrapper()
                {
                    WrappedTarget = target,
                    DefaultFilter = "level >= LogLevel.Info", // by default log info and above
                };
                wrapper.Initialize(null);
                target.Initialize(null);
                var exceptions = new List<Exception>();
                wrapper.WriteAsyncLogEvent(new LogEventInfo(LogLevel.Info, "Logger1", "Hello").WithContinuation(exceptions.Add));
                Assert.Single(target.Events);
                wrapper.WriteAsyncLogEvent(new LogEventInfo(LogLevel.Debug, "Logger1", "Hello").WithContinuation(exceptions.Add));
                Assert.Single(target.Events);
            }
        }
    }
}

File: refactored/duplicate-assert/python-pytest/PostFilteringTargetWrapperOnlyDefaultFilter.py
import pytest

class TestPostFilteringTargetWrapper:

    def test_post_filtering_target_wrapper_only_default_filter1(self):
        target = MyTarget()
        wrapper = PostFilteringTargetWrapper()
        wrapper.set_wrapped_target(target)
        wrapper.set_default_filter("level >= LogLevel.Info")  # by default log info and above
        wrapper.initialize(None)
        target.initialize(None)
        exceptions = []
        wrapper.write_async_log_event(LogEventInfo(LogLevel.Info, "Logger1", "Hello").with_continuation(exceptions.append))
        assert len(target.get_events()) == 1

    def test_post_filtering_target_wrapper_only_default_filter2(self):
        target = MyTarget()
        wrapper = PostFilteringTargetWrapper()
        wrapper.set_wrapped_target(target)
        wrapper.set_default_filter("level >= LogLevel.Info")  # by default log info and above
        wrapper.initialize(None)
        target.initialize(None)
        exceptions = []
        wrapper.write_async_log_event(LogEventInfo(LogLevel.Info, "Logger1", "Hello").with_continuation(exceptions.append))
        wrapper.write_async_log_event(LogEventInfo(LogLevel.Debug, "Logger1", "Hello").with_continuation(exceptions.append))
        assert len(target.get_events()) == 1
